### PR TITLE
build everything with ghc-98

### PR DESCRIPTION
**Pull request by lucasdicioccio**

### CODE REVIEW

File: http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
-import           Data.Binary.Get (getByteString, getInt8, getWord32be, runGetIncremental)
+import Data.Binary.Builder (fromByteString, putWord32be, singleton)
+import Data.Binary.Get (getByteString, getInt8, getWord32be, runGetIncremental)
 import qualified Data.ByteString.Char8 as ByteString
-import           Data.ProtoLens.Encoding (encodeMessage, decodeMessage)
-import           Data.ProtoLens.Message (Message)
-import           Data.ProtoLens.Service.Types (Service(..), HasMethod, HasMethodImpl(..))
-import           Data.Proxy (Proxy(..))
-import           GHC.TypeLits (Symbol, symbolVal)
+import Data.Kind (Type)
```
Comment (lucasdicioccio) [inline http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs:15]: probably needs some CPP to be compatible with old GHCs

File: http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
 import Data.Monoid ((<>))
 #endif
 
-import Network.GRPC.HTTP2.Types
 import Network.GRPC.HTTP2.Encoding
+import Network.GRPC.HTTP2.Types
 
 -- | A proxy type for giving static information about RPCs.
-data RPC (s :: *) (m :: Symbol) = RPC
+data RPC (s :: Type) (m :: Symbol) = RPC
```
Comment (lucasdicioccio) [inline http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs:31]: probably needs some CPP to be compatible with old GHCs

**Comment by lucasdicioccio** [comment]
@ProofOfKeags WDYT? looks like we're getting there

File: http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs [hunk 3]
```diff
@@ -28,7 +31,11 @@ import Network.GRPC.HTTP2.Encoding
 import Network.GRPC.HTTP2.Types
 
 -- | A proxy type for giving static information about RPCs.
+#if MIN_VERSION_base(4,11,0)
 data RPC (s :: Type) (m :: Symbol) = RPC
+#else
+data RPC (s :: *) (m :: Symbol) = RPC
```
Comment (ProofOfKeags) [inline http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs:37]: Is `NoStarIsType` standard now?

**Comment by ProofOfKeags** [comment]
I haven't pulled it down and built it/tested, but the changes are uncontroversial.

File: http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs (hunk 3 above)
Comment (lucasdicioccio) [reply http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs:37]: I got a warning without it and mechanically cleared it without afterthought. Warning, I guess, means stars will be deprecated at some point (I haven't followed through TBH).

**Comment by lucasdicioccio** [comment]
> I haven't pulled it down and built it/tested, but the changes are uncontroversial.

Let's merge it before I go into another tunnel.

File: http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs (hunk 3 above)
Comment (ProofOfKeags) [reply http2-grpc-proto-lens/src/Network/GRPC/HTTP2/ProtoLens.hs:37]: That's fine, I prefer enabling `NoStarIsType` anyway in my own code. I was just curious.

File: http2-grpc-proto3-wire/http2-grpc-proto3-wire.cabal [hunk 4]
```diff
@@ -41,6 +41,6 @@ library
     , bytestring >=0.10.8 && <0.12
```
Comment (alexfmpe) [inline http2-grpc-proto3-wire/http2-grpc-proto3-wire.cabal:41]: Doesn't `bytestring` here (and in the other cabal files) also need `< 0.13` like above in bench-tool ?

**Pull request merged**